test/Memory.cpp

```diff
@@ -5,6 +5,24 @@
 #include <numeric>
 #include <vector>
 
+TEST_CASE("operator new", "[memory]") {
+    SECTION("allocates memory") {
+        auto ptr = new int32_t;
+        REQUIRE(ptr != nullptr);
+
```

Comment:
writing to it as well would make sense for a test like this, but failure is not catchable with catch2